tanjun/checks.py

```diff
@@ -509,6 +511,43 @@ async def __call__(
         return self._handle_result((permissions & self._permissions) == self._permissions)
 
 
+class HasAnyRoleCheck(_Check):
+    __slots__ = ("required_roles", "ids_only")
+
+    def __init__(
+        self,
+        roles: collections.Sequence[typing.Union[hikari.SnowflakeishOr[hikari.Role], str]] = [],
+        *,
+        error_message: typing.Optional[str] = "You do not have the required roles to use this command!",
+        halt_execution: bool = True,
+    ) -> None:
+        super().__init__(error_message, halt_execution)
+        self.required_roles = roles
```

Comment:
Another style thing, these attributes should be private now (so `_required_roles` and `_ids_only`